### ISSUE TITLE
What power supply do you use for the Ardruino?

### ISSUE THREAD

**Issue by hkchantmhk** [night-howler/protogasm#14]
I'm ordering parts to build my own but I came across an issue about the power supply for the Ardruino. It can take 7-12V. So what voltage did you use and at what amperage? Does it even matter?

Many thanks.

**Comment by Gollam** [comment]
@hkchantmhk What did you end up using?

**Comment by Nuk3d02** [comment]
Was also curious about this... I have everything built and was able to test the board with USB, but I'm sure that won't be sufficient to power the vibe. I was thinking of just going with a 12v supply, and I'm fairly sure as long as it's at least 1 amp?

**Comment by Gollam** [comment]
@Nuk3d02 are we supposed to use one power supply for the whole thing? I'm trying to understand the schematic, and I can see multiple VCC and VIN connections.

**Comment by Nuk3d02** [comment]
@Gollam I assumed it's one supply for the whole thing... They didn't mention adding power anywhere else and I feel like a second power source could fry the arduino but I'm not that knowledgeable about electronics.

**Comment by Gollam** [comment]
@Nuk3d02 I'll let you know if I figure out anything. BTW what material did you choose to print the parts from?

**Comment by Nuk3d02** [comment]
@Gollam I just did PLA. I think it's good enough for this

I just got the motor this past afternoon, so I'll test it out today and I'll let you know what I find too

**Comment by Gollam** [comment]
All right, my motor is still under shipment.

**Comment by Nuk3d02** [comment]
@Gollam So yeah, just using a 5v usb brick powers everything fine but I just feel like the motor could benefit from a higher voltage. it's no magic wand, but maybe it'll still do the job

**Comment by nogasm** [comment]
A 5V brick will be enough to power the LEDs and microcontroller, but not the vibrator motor. The original nogasm used a 12V 3A power supply - I think that should be enough in this case too.

**Comment by Gollam** [comment]
@nogasm thanks for the help :)

**Comment by Nuk3d02** [comment]
@nogasm yes, but the arduino can only output max 5v and we're powering the arduino itself right?
I didn't see anything where we put power into the board

I know I don't understand a whole lot about electronics like this, but I think I have that basic understanding

**Comment by nogasm** [comment]
There's a MOSFET that drives the motor at higher voltages, since the arduino alone isn't powerful enough to. Arduinos operate at 5V but have a separate power input circuit that accepts 7-12V. In this design, that one 12V circuit called Vin is used both to drive the motor, and power the arduino.

**Comment by Nuk3d02** [comment]
Ohh, I see now, I had a bad assumption. Also hadn't googled correctly. Thank you!

**Comment by logz1** [comment]
Need help regarding this same question. From the research i found both the power jack and the VIN on the Arduino uno are limited in voltage and amperage. The motor i am using is rated for 12 volts and 2.4 amps. If i connect a 12 volt 3 amp psu to the VIN or the power jack, wont that fry the Arduino? Additionally if it cant be powered through the jack because of amp draw limitations, how would a psu be connected safely to the VIN and ground pins? Any Photos of a completed power adapter would be greatly appreciated!

**Comment by Nuk3d02** [comment]
@logz1 I thought the same thing. But you can actually safely power both from the dc jack, the arduino is actually rated up to 12v.

**Comment by logz1** [comment]
Ah ok. So i can use a standard 12 volt psu at 1 amp or should it be rated at 3 amps as suggested in a previous comment by nogasm? Thanks!

**Comment by Nuk3d02** [comment]
3A would be best I think.

**Comment by Gollam** [comment]
@logz1 don't worry about the current, 3A is the maximum which the PSU can provide, that does not mean that it will deliver 3A all the time, it will only give what the circuit demands. The voltage however has to be correct.

**Comment by hkchantmhk** [comment]
[EDIT] A 12V wall plug to the barrel jack turned out to be problematic. The board wasn't able to handle the voltage for some rason. Everything worked when powered via the USB, but stopped working altogether when using 12V from the barrel jack.

I am going to get a buck converter to give me 5V for the board but continue to power the motor on 12V (it's a 12V motor). 5V gives a very sluggish motor and 5V motors don't vibrate strong enough for the purpose IMHO.

> Sorry I kind of forgot about this. Ended up using a 12V 3A wall plug to power the arduino.
